lib/core/src/sdk.rs

```diff
@@ -1674,3 +1674,306 @@ impl ReconnectHandler for SwapperReconnectHandler {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
+
+    use anyhow::{anyhow, Result};
+    use boltz_client::{
+        boltzv2::{self, SwapUpdateTxDetails},
+        swaps::boltz::{ChainSwapStates, RevSwapStates, SubSwapStates},
+    };
+    use lwk_wollet::{elements::encode::serialize, hashes::hex::DisplayHex};
+
+    use crate::{
+        model::{Direction, PaymentState, Swap},
+        sdk::LiquidSdk,
+        test_utils::{
+            chain_swap::new_chain_swap,
+            create_mock_tx,
+            persist::{new_persister, new_receive_swap, new_send_swap},
+            sdk::new_liquid_sdk,
+            status_stream::MockStatusStream,
+            swapper::MockSwapper,
+        },
+    };
+    use paste::paste;
+
+    macro_rules! trigger_swap_update {
```

Comment:
What is the benefit here for macro instead of a function?

Reply:
We can't use a function because the return type differs between each kind. It's possible but would be more verbose to implement and we would have to pattern match the `Swap` result within the tests